src/hooks/useMerklRewardsOnCampaigns.ts

```diff
@@ -0,0 +1,190 @@
+  const activeCampaigns = [] as string[];
+  const pastCampaigns = [] as string[];
+
+  // Call to get the active positions
+  // To do -> use the label to get only
+  const MERKL_POSITIONS_API = `${MERKL_API}/multiChainPositions?chainIds=${ACTIVE_CHAINS.join(',')}&user=${userAddress}`; //&creatorTag=${CREATOR_TAG}
+  const {
+    data: positionsData,
+    isSuccess: positionsIsSuccess,
+    isLoading: positionsIsLoading,
+  } = useQuery({
+    queryKey: ['MerklPositions'],
+
+    queryFn: async () => {
+      const response = await fetch(MERKL_POSITIONS_API, {});
+      const result = await response.json();
+      return result as MerklPositionData;
+    },
+    enabled: !!userAddress,
+    refetchInterval: 1000 * 60 * 60,
+  });
+
+  // loop through the chains and positions
+  if (positionsData) {
+    for (const chain of ACTIVE_CHAINS) {
+      if (positionsData[chain]) {
+        for (const [key, data] of Object.entries(positionsData[chain])) {
+          activeCampaigns.push(key);
+        }
+      }
+    }
+  }
+
+  // check the user positions for the interesting campaign
+  const MERKL_REWARDS_API = `${MERKL_API}/rewards?chainIds=${ACTIVE_CHAINS.join(',')}&user=${userAddress}`;
+  const {
+    data: rewardsData,
+    isSuccess: rewardsIsSuccess,
+    isLoading: rewardsIsLoading,
+  } = useQuery({
+    queryKey: ['MerklRewards'],
+
+    queryFn: async () => {
+      const response = await fetch(MERKL_REWARDS_API, {});
+      const result = await response.json();
+      return result;
+    },
+    enabled: !!userAddress,
+    refetchInterval: 1000 * 60 * 60,
+  });
+
+  // transform to know what is not coming from Jumper campaigns
+
+  if (rewardsData) {
+    for (const rewardElem of REWARDS_LIST) {
+      const tokenData = rewardsData[rewardElem.chainId]?.tokenData;
+      if (tokenData) {
+        rewardsToClaim = rewardsToClaim.concat(
+          Object.entries(tokenData)
+            .map((elem): AvailableRewards => {
+              const key = elem[0];
+              const value = elem[1] as TokenData;
+              return {
+                chainId: parseInt(rewardElem.chainId),
+                address: key,
+                symbol: value.symbol,
+                accumulatedAmountForContractBN: value.accumulated,
+                amountToClaim: (value.unclaimed as any) / 10 ** value.decimals, //todo: need to be typed with big int
+                amountAccumulated:
+                  (value.unclaimed as any) / 10 ** value.decimals, //todo: need to be typed with big int
+                proof: value.proof,
+                explorerLink: rewardElem.explorerLink,
+                tokenLogo: rewardElem.lightTokenLogo,
+                chainLogo: rewardElem.lightChainLogo,
+                claimingAddress: rewardElem.claimingAddress,
+                decimalsToShow: rewardElem.decimalsToShow,
+              };
+            })
+            .filter(
+              (elem) =>
+                elem.address.toLowerCase() ===
```

Comment:
Used multiple time across the app, would be good to have a helper and refactorize it

Reply:
Creating a ticket for this to do in next sprint.